be/src/util/timezone_utils.cpp

```diff
@@ -63,6 +63,172 @@ void TimezoneUtils::load_timezone_names() {
+        diff_seconds = *reinterpret_cast<const int32_t*>(arg.tt_utoff + 0);
+        is_dst = arg.tt_isdst;
+        name_index = arg.tt_desigidx;
+    }
+
+    int32_t diff_seconds; // to UTC
+    bool is_dst;
+    uint8_t name_index;
+};
+
+template <>
+ttinfo next_from_charstream<ttinfo>(int8_t*& src) {
+    ttinfo value = *reinterpret_cast<ttinfo*>(src);
+    src += TTINFO_SIZE;
+    if constexpr (std::endian::native == std::endian::little) {
+        std::swap(value.tt_utoff[0], value.tt_utoff[3]);
+        std::swap(value.tt_utoff[1], value.tt_utoff[2]);
+    }
+    return value;
+}
+
+/* 
+ * follow the rule of tzfile(5) which defined in https://man7.org/linux/man-pages/man5/tzfile.5.html.
+ * should change when it changes.
+ */
+bool parse_load_timezone(vectorized::ZoneList& zone_list, int8_t* data, int len,
+                         bool first_time = true) {
+    int8_t* begin_pos = data;
+    /* HEADERS */
+    if (memcmp(data, "TZif", 4) != 0) [[unlikely]] { // magic number
+        return false;
+    }
+    data += 4;
+
+    // if version = 2, the whole header&data will repeat itself one time.
+    int8_t version = next_from_charstream<int8_t>(data) - '0';
+    data += 15; // null bits
+    int32_t ut_count = next_from_charstream<int32_t>(data);
+    int32_t wall_count = next_from_charstream<int32_t>(data);
+    int32_t leap_count = next_from_charstream<int32_t>(data);
+    int32_t trans_time_count = next_from_charstream<int32_t>(data);
+    int32_t type_count = next_from_charstream<int32_t>(data);
+    int32_t char_count = next_from_charstream<int32_t>(data);
+
+    /* HEADERS end, FIELDS begin*/
+    // transaction time points, which we don't need
+    data += (first_time ? 5 : 9) * trans_time_count;
+
+    // timezones
+    std::vector<real_ttinfo> timezones(type_count);
+    for (int i = 0; i < type_count; i++) {
+        ttinfo tz_data = next_from_charstream<ttinfo>(data);
+        timezones[i] = tz_data; // cast by c'tor
+    }
+
+    // timezone names
+    const char* name_zone = (char*)data;
+    data += char_count;
+
+    // concate names
+    for (auto& tz : timezones) {
+        int len = strlen(name_zone + tz.name_index);
+        zone_list.emplace(std::string {name_zone + tz.name_index, name_zone + tz.name_index + len},
+                          cctz::fixed_time_zone(cctz::seconds(tz.diff_seconds)));
+    }
+
+    // the second part.
+    if (version == 2 && first_time) {
+        // leap seconds, standard/wall indicators, UT/local indicators, which we don't need
+        data += 4 * leap_count + wall_count + ut_count;
+
+        return (data < begin_pos + len) &&
+               parse_load_timezone(zone_list, data, len - (data - begin_pos), false);
+    }
+
+    return true;
+}
+
+} // namespace
+
+void TimezoneUtils::load_timezones_to_cache(vectorized::ZoneList& cache_list) {
+    std::string base_str;
+    const char* tzdir = "/usr/share/zoneinfo"; // default
+    // try get from System
```

Comment:
you should make sure, both macos and linux are using /usr/share/zoneinfo this file to save timezone info.

Reply:
yes, for macos and linux that's true.